provider/server/server_suite_test.go

```diff
@@ -38,38 +47,97 @@ func TestServer(t *testing.T) {
+	logf.SetLogger(GinkgoLogr)
 
-	machineClasses, err := mcr.NewMachineClassRegistry([]iri.MachineClass{
+	machineClasses := []iriv1alpha1.MachineClass{
 		{
-			//TODO
 			Name: machineClassx3xlarge,
-			Capabilities: &iri.MachineClassCapabilities{
+			Capabilities: &iriv1alpha1.MachineClassCapabilities{
 				CpuMillis:   4000,
 				MemoryBytes: 8589934592,
 			},
 		},
-	})
+	}
+	machineClassData, err := json.Marshal(machineClasses)
+	Expect(err).NotTo(HaveOccurred())
+
+	machineClassesFile, err := os.CreateTemp(GinkgoT().TempDir(), "machineclasses")
 	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		_ = machineClassesFile.Close()
+	}()
+	Expect(os.WriteFile(machineClassesFile.Name(), machineClassData, 0666)).To(Succeed())
 
-	srv, err = server.New(server.Options{
-		BaseURL:        baseURL,
-		MachineStore:   machineStore,
-		MachineClasses: machineClasses,
-	})
+	By("starting the app")
+
+	userHomeDir, err := os.UserHomeDir()
 	Expect(err).NotTo(HaveOccurred())
+
+	fs := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	apinetPlugin := networkinterfaceplugin.NewDefaultOptions()
+	apinetPlugin.PluginName = "apinet"
+	apinetPlugin.AddFlags(fs)
+	Expect(fs.Set("apinet-node-name", "test-node")).NotTo(HaveOccurred())
+
+	opts := app.Options{
+		//TODO: set these using env variables if needed
+		Address:                     fmt.Sprintf("%s/test.sock", os.Getenv("PWD")),
+		BaseURL:                     baseURL,
+		PathSupportedMachineClasses: machineClassesFile.Name(),
+		RootDir:                     fmt.Sprintf("%s/libvirt-provider", userHomeDir),
+		StreamingAddress:            streamingAddress,
+		Libvirt: app.LibvirtOptions{
+			Socket:                "/var/run/libvirt/libvirt-sock",
+			URI:                   "qemu:///system",
+			PreferredDomainTypes:  []string{"kvm", "qemu"},
+			PreferredMachineTypes: []string{"pc-q35", "pc-i440fx-2.9", "pc-i440fx-2.8"},
+			Qcow2Type:             "exec",
+		},
+		NicPlugin: apinetPlugin,
+		// TODO: add other required fields
+	}
+
+	srvCtx, cancel := context.WithCancel(context.Background())
+	DeferCleanup(cancel)
+
+	go func() {
+		defer GinkgoRecover()
+		Expect(app.Run(srvCtx, opts)).To(Succeed())
+	}()
+
+	Eventually(func() (bool, error) {
+		return isSocketAvailable(opts.Address)
+	}, "30s", "500ms").Should(BeTrue(), "The UNIX socket file should be available")
+
+	address, err := machine.GetAddressWithTimeout(3*time.Second, fmt.Sprintf("unix://%s", opts.Address))
+	Expect(err).NotTo(HaveOccurred())
+
+	gconn, err := grpc.Dial(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	Expect(err).NotTo(HaveOccurred())
+
+	machineClient = iriv1alpha1.NewMachineRuntimeClient(gconn)
+	DeferCleanup(gconn.Close)
+
+	c := dialers.NewLocal()
+	libvirtConn = libvirt.NewWithDialer(c)
+	err = libvirtConn.Connect()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(libvirtConn.IsConnected(), BeTrue())
+
+	DeferCleanup(libvirtConn.ConnectClose)
 })
+
+func isSocketAvailable(socketPath string) (bool, error) {
```

Comment:
do you need boolean? You can return error, if socket is missing and otherwise it is nil